reduction_scripts/reduce_data.py

```diff
@@ -896,99 +912,190 @@ def run_save_3dcube(metadata, prev_suffix, curr_suffix, **args):
+    # Match cubes from the same observation based on DATE-OBS 
+    # ----------------------------------------------------------
+
+    matched_list = cube_matcher(reduced_cubes_paths)
+
+    # ----------------------------------------------------------
+    # Read extraction parameters from JSON file
+    # ----------------------------------------------------------
+
+    # Read the JSON file and define parameters
+    file_path = os.path.join(reduction_scripts_dir, f"params_extract_{obs_mode}.json")
+
+    with open(file_path, "r") as f:
+        extract_params = json.load(f)
+    
+    # Extract parameters
+    sky_sub = extract_params["sky_sub"]
+    check_plot = extract_params["check_plot"]
+    border_width = extract_params["border_width"]
+    r_arcsec = extract_params["r_arcsec"]
+
+    # ----------------------------------------------------------
+    # Loop over matched cubes list
+    # ----------------------------------------------------------
+
+    for match_cubes in matched_list:
+        # ----------
+        # Extraction
+        # ----------
+        blue_cube_path = match_cubes["Blue"]
+        red_cube_path = match_cubes["Red"]
+        plot_output = match_cubes['file_name'] + '_detected_apertures_plot.pdf'
+
+        # Run auto-extraction
+        auto_extract(
+            blue_cube_path,
+            red_cube_path,
+            destination_dir,
+            r_arcsec=r_arcsec,
+            border_width=border_width,
+            sky_sub=sky_sub,
+            check_plot=check_plot,
+            plot_output=plot_output
+        )
+        
+        # ------------------------------------
+        # Splice only paired cubes and spectra
+        # ------------------------------------
+        if match_cubes["Blue"] is not None and match_cubes["Red"] is not None:
+            # Splice cubes
+            blue_cube_name = os.path.basename(match_cubes["Blue"])
+            spliced_cube_name = blue_cube_name.replace("Blue", "Splice")
+            spliced_cube_path = os.path.join(destination_dir, spliced_cube_name)
+            splice_cubes(match_cubes["Blue"], match_cubes["Red"], spliced_cube_path)
+
+            # Splice spectra
+            pattern_blue = os.path.join(
+                destination_dir, blue_cube_name.replace("cube", "spec.ap*")
+            )
+            pattern_red = os.path.join(
+                destination_dir,
+                os.path.basename(match_cubes["Red"]).replace("cube", "spec.ap*"),
+            )
+
+            blue_specs = glob.glob(pattern_blue)
+            red_specs = glob.glob(pattern_red)
+
+            for blue_spec, red_spec in zip(blue_specs, red_specs):
+                spliced_spectrum_name = os.path.basename(blue_spec).replace(
+                    "Blue", "Splice"
+                )
+                output = os.path.join(
+                    working_dir, destination_dir, spliced_spectrum_name
+                )
+                splice_spectra(blue_spec, red_spec, output)
```

Comment:
We have tried this, but the extraction-splice workflow doesn't fit well within the reduction structure. For example, we have to apply the same extraction parameters to both arms, so these parameters should be repeated in both the blue and red arms in the JSON file, which makes things less clear for users. Also, implementing these routines within the step structure makes the code less clear. Extraction and splicing are technically not part of the reduction process. Therefore, one option here is to put all this functionality in a dedicated module.

Reply:
Yeah, a dedicated module would make sense, it will depend on what the users want. This whole script could be broken into separate modules with separate configs if needed. Then a much simpler 'reduce_data_and_splice.py' could run the whole process, for example.

Reply:
We could split the code in modules in following PR, for example when we implement the structure the archive team wants .